dcos/http.py

```diff
@@ -25,11 +33,12 @@ def _default_is_success(status_code):
 
 
 @util.duration
-def request(method,
-            url,
-            is_success=_default_is_success,
-            timeout=None,
-            **kwargs):
+def _request(method,
+             url,
+             is_success=_default_is_success,
+             timeout=None,
+             realm=None,
```

Comment:
`realm` is a strange parameter to send.  Can we just make this `auth`, and do the lookup in the caller?